src/website/views/components/TimestampFilter/TimestampFilter.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import Accordion from '@mui/material/Accordion';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Typography from '@mui/material/Typography';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { Button, Grid } from '@mui/material';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import styles from './timestampfilter.module.css';
+import DataFilterActions from '@/stores/DataFilter/DataFilterActions';
+
+function TimestampFilter(props) {
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+
+  const dispatch = props.dispatch;
+
+  const handleApplyChange = () => {
+    let emptyDate = startDate != null && endDate != null;
+    let validDate = parseISOString(endDate) > parseISOString(startDate);
+
+    if (emptyDate == true && validDate == true) {
+      dispatch({
+        type: DataFilterActions.SET_TIMESTAMP,
+        payload: {
+          startDate: startDate.toISOString(),
+          endDate: endDate.toISOString(),
+        },
+      });
```

Comment:
Rather than calling the dispatch function here, let's define it in a more structured way. Have an object called `mapDispatchToProps` below. Then define the action we wish to dispatch in there: 

Here's an example:
```
const mapDispatchToProps = {
  setTimeStamp: (timestamps: any) => {
    return {
      type: DataFilterActions.SET_TIMESTAMP,
      payload: timestamps
    }
  }
}

export default connect(mapDispatchToProps)(TimestampFilter);
```
Then you can pass `setTimeStamp` to `TimeStampFilter` as a prop (e.g. ` TimestampFilter({ setTimeStamp })`) and call the function itself to dispatch an action. That is, instead of writing `dispatch({...})`, you can just call `setTimeStamp()` now to dispatch the action.

Please apply the same change to all other dispatch calls below

Reply:
done